### ISSUE TITLE
Feature reuquest: Investigate replacing the mainboard with a RasPi

### ISSUE THREAD

**Issue by MarcusWolschon** [PostalBlab/hackaverde#1]
Could it be possible to not use the Mainboard CPU?

1. Measure timing information on the PORT D heater pin for light, medium and dark roasting profile
2. Connect a Raspberry Pi to the TRIACS, heater and sensor inputs?
3. Still use the power supply on the main board.
4. Run our own software, controled via console/Wifi+app/touchscreen/whatever?

**Comment by PostalBlab** [comment]
That is of course possible, but the machine has to be taken apart which is a pain in the ass and not doable by everyone. But if you want to do it, i can share the disassembly infos i have.

**Comment by MarcusWolschon** [comment]
For replacing the firmware it has to be taken apart too, doesn't it?

**Comment by PostalBlab** [comment]
No, the SWD port is accessible from the front after removing the cover on the touch/rfid panel. It is visible in this image: https://github.com/PostalBlab/hackaverde/blob/master/images/exposed_swd.jpg

**Comment by MarcusWolschon** [comment]
That SWD port sounds like a great way to attach an ESP32 or a RasPi.
So next steps: Find out the pinout of that header and attach a logic probe before scanning well known NFC cards.

**Comment by MarcusWolschon** [comment]
I got myself a second bonaverde for 50€ to take apart, attach probes to sensors and actuators and replace the entire main PCB (keeping the power pcb) with a Raspberry Pi+touchscreen or similar just operating the relays and reading the sensors.

**Comment by MarcusWolschon** [comment]
I have started a separate Github repository for doing the Raspberry Pi conversion.
(Will upload later tonight)
I already did the disassembly-instructions and am looking at the electronics right now.
https://github.com/MarcusWolschon/raspiverde

Looks like a the inputs to the "power_control_pcb" from the front panel could be replaced
without having to attach anything to the well housed high-voltage or high-power parts.
